src/services_models/services_constructor.jl

```diff
@@ -30,8 +30,12 @@ function construct_service!(
 ) where {SR <: PSY.Reserve}
     services_mapping = PSY.get_contributing_device_mapping(sys)
     time_steps = model_time_steps(psi_container)
-    names = (PSY.get_name(s) for s in services)
-
+    names = [
```

Comment:
I understand why these changes fix the problem.  However, this is a non-trivial amount of logic that I suspect other users will have to replicate and possibly forget or mess up when they make similar functions.  Is that true?  If so, would it be better to filter out these services in the place where this function is called?

Reply:
That's why I suggested to not create the entry in the output of `PSY.get_contributing_device_mapping`